inst/include/mode/mode.hpp

```diff
@@ -31,6 +31,10 @@ class container {
     return solver_[0].time();
   }
 
+  std::vector<double> solve(int end_time) {
```

Comment:
So the simplest interface for that would involve passing a vector of y0 into `solve`, one for each particle, and getting back a vector of y values at `end_time`?

Reply:
It would end up being a matrix, but we'd need to move away from a nice vector -> vector approach and move to using something like iterators - see https://github.com/mrc-ide/dust/blob/master/inst/include/dust/particle.hpp#L33-L38 and https://github.com/mrc-ide/dust/blob/master/inst/include/dust/particle.hpp#L74-L78 (equivalent to our solver.hpp).

Also see https://github.com/mrc-ide/dust/blob/master/inst/include/dust/r/dust.hpp#L221-L236 for the equivalent to solve() (we might adopt `run` here as the method name actually)